### PR TITLE
Remove mssql dependency, use tediousjs/connection-string for parsing

### PR DESCRIPTION
**Pull request by zijchen**
Remove mssql dependency since we only use it for parsing connection string now that everything else is using go-sqlcmd to test connection. Now we use [tediousjs/connection-string](https://github.com/tediousjs/connection-string) for parsing, which was what mssql was using under the hood

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -16,6 +16,9 @@ inputs:
   build-arguments:
     description: 'In case of a .sqlproj file, additional arguments that will be applied to dotnet build when building the database project.'
     required: false
+  client-id:
```
Comment (zijchen) [inline action.yml:19]: remove

**Review by dzsquared — DISMISSED** [review]

File: src/SqlConnectionConfig.ts [hunk 2]
```diff
@@ -60,150 +64,123 @@ export default class SqlConnectionConfig {
...
         }
 
-        if (this._connectionConfig.password) {
-            core.setSecret(this._connectionConfig.password);
+        if (this._parsedConnectionString.password) {
+            core.setSecret(this._parsedConnectionString.password);
         }
     }
 
+    private _parseConnectionString(connectionString: string): SqlConnectionString {
```
Comment (chlafreniere) [inline src/SqlConnectionConfig.ts:76]: Can you look through https://github.com/tediousjs/connection-string to see if there's any logic that we're missing? Anywhere where we're parsing connection strings makes me a bit paranoid 😰.

Also curious if you took a look at leveraging that library as a dependency. Looks like it's pretty widely used: https://www.npmjs.com/package/@tediousjs/connection-string

Comment (zijchen) [reply src/SqlConnectionConfig.ts:76]: I did look into that library, it is what was used under the hood in v2 for parsing. I just switched to the old regex parsing logic from v1.

The parsing library has no external dependencies so we can take it if you want.

Comment (chlafreniere) [reply src/SqlConnectionConfig.ts:76]: I think I'd rather use that that have our homegrown connection string parsing, to be honest. Would you mind making the change, if it's not too much work?

Comment (zijchen) [reply src/SqlConnectionConfig.ts]: Updated to use that library instead

**Review by chlafreniere — APPROVED** [review]

**Pull request merged**